util/visualizer.py

```diff
@@ -245,3 +245,35 @@ def display_img(self,img_path):
         im = np.transpose(im, (2,0,1))
         img_name=img_path.split('/')[-1].split('.')[0]
         self.vis.image(im,opts=dict(title=self.name + ' ' + img_name))
+
+    def plot_current_fid(self, epoch, counter_ratio, fids):
+        """display the current fid values on visdom display: dictionary of fid labels and values
+
+        Parameters:
+            epoch (int)           -- current epoch
+            counter_ratio (float) -- progress (percentage) in the current epoch, between 0 to 1
+            fids (OrderedDict)  -- training fid values stored in the format of (name, float) pairs
+        """
+        if not hasattr(self, 'plot_fid'):
+            self.plot_fid = {'X': [], 'Y': [], 'legend': list(fids.keys())}
+        self.plot_fid['X'].append(epoch + counter_ratio)
+        if len(self.plot_fid['legend']) == 1 :
+            self.plot_fid['Y'].append(fids[self.plot_fid['legend'][0]])
+            X=np.array(self.plot_fid['X'])
+            Y=np.array(self.plot_fid['Y'])
+        else :
+            self.plot_fid['Y'].append([fids[k] for k in self.plot_fid['legend']])
+            X=np.stack([np.array(self.plot_fid['X'])] * len(self.plot_fid['legend']), 1)
+            Y=np.array(self.plot_fid['Y'])
```

Comment:
I think the code for len(legend) > 1 works for len(legend) == 1 too (you just call stack on one array). So this could be simplified here

Reply:
Done :)